muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/forum/rest/ForumRESTService.java

```diff
@@ -898,16 +904,34 @@ public Response listThreadSubscriptionsByUser(@PathParam ("USERID") Long userId)
       if (userEntityController.isStudent(sessionController.getLoggedUserEntity()) && !sessionController.getLoggedUserEntity().getId().equals(userEntity.getId())) {
         return Response.status(Status.FORBIDDEN).entity("You can list your own subscriptions only").build();
       }
-      
       return Response.ok(createThreadSubscriptionRestModel(forumThreadSubscriptionController.listByUser(userEntity).toArray(new ForumThreadSubscription[0]))).build();
     } else
       return Response.status(Status.FORBIDDEN).build();
   }
   
   private ForumThreadSubscriptionRESTModel createThreadSubscriptionRestModel(ForumThreadSubscription entity) {
+    
+    Long workspaceEntityId = null;
+    String workspaceUrlName = null;
+    String workspaceName = null;
+    
+    if (entity.getForumThread() != null) {
+      WorkspaceForumArea workspaceForumArea = forumController.findByAreaId(entity.getForumThread().getForumArea().getId());
+      
+      if (workspaceForumArea != null) {
+        WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(workspaceForumArea.getWorkspace());
+        if (workspaceEntity != null) {
+          Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
```

Comment:
Unnecessary call to Pyramus to resolve workspace name. Use...

`workspaceEntityController.getName(workspaceEntity);`

...instead.